scripts/move_patterns.py

```diff
@@ -0,0 +1,268 @@
+       # If you only want to do list once exit after first for loop
+       while (True):
+           for waypoint in waypointList:
+               x,y,yaw,comment = waypoint
+               x = (x * scaleX) + offsetX
+               y = (y * scaleY) + offsetY
+               now = rospy.get_rostime()
+               print "[%i.%i]  Waypoint: %s X %f Y %f yaw %f  will be published now" \
+                     % (now.secs,now.nsecs,comment,x, y, yaw)
+
+               # now publish the waypoint
+               moveResult = self.publishMoveBaseGoalWaitForReply( x,  y, yaw, comment)
+               if moveResult == True:
+                   print "ERROR: MoveBasic Bad Status! for Waypoint: X %f Y %f yaw %f " % (x, y, yaw)
+               else:
+                   print "[%i.%i]  Waypoint: %s X %f Y %f yaw %f  has been reached" \
+                         % (now.secs,now.nsecs,comment,x, y, yaw)
+
+               # optionally wait at each vertex before going to next one
+               if (waitAtEachVertex == 1):
+                   raw_input("Hit ENTER to go to next waypoint ... ")
+
+
+
+    # A publisher for sending commands to the follower node
+    # self.moveBaseGoalPub = rospy.Publisher("/move_base/goal", \
+    #     move_base_msgs/MoveBaseActionGoal, queue_size=1)
+
+    # Publish the desired waypoint to navigate on top of now
+    # For fiducial nav use frame_id of  "map"  
+    # For odom nav use frame_id of "odom"
+    def publishMoveBaseGoalWaitForReply(self, x, y, yaw, comment):
+        goal = MoveBaseGoal()
+        goal.target_pose.header.frame_id = "map"     
+        goal.target_pose.header.stamp = rospy.Time.now()
+        goal.target_pose.pose.position.x = x 
+        goal.target_pose.pose.position.y = y 
+
+        # to send orientation with a yaw we need quaternion transform
+        x , y, z, w = tf.transformations.quaternion_from_euler(0, 0, yaw)
+        goal.target_pose.pose.orientation.x = x
+        goal.target_pose.pose.orientation.y = y
+        goal.target_pose.pose.orientation.z = z
+        goal.target_pose.pose.orientation.w = w
+        now = rospy.get_rostime()
+        print "[%i.%i]  PubMove:  %s x,y,z,w of %f %f %f %f yaw %f" \
+              % (now.secs,now.nsecs,comment,x,y,z,w,yaw)
+
+        client = actionlib.SimpleActionClient('move_base', MoveBaseAction)
+        client.wait_for_server()
+
+        # publish the goal to the topic
+        client.send_goal(goal)
+
+        now = rospy.get_rostime()
+        print "[%i.%i]  Waiting for result ..." % (now.secs, now.nsecs)
+        wait = client.wait_for_result()
+        if not wait:
+            rospy.logerr("Action server not available!")
+            rospy.signal_shutdown("Action server not available!")
+        else:
+            now = rospy.get_rostime()
+            print "[%i.%i]  Received result" % (now.secs, now.nsecs)
+            return client.get_result()
+
+    """
+    Main loop
+    """
+    def run(self):
```

Comment:
It looks like `run()` doesn't actually do anything

Reply:
see next reply